services/server/test/integration/repository-handlers/files.spec.ts

```diff
@@ -43,46 +40,84 @@ describe("Verify repository endpoints", function () {
     chai.expect(res4.body.full).has.a.lengthOf(1);
   });
 
-  it("should handle pagination in /files/contracts/{chain}", async function () {
-    for (let i = 0; i < 5; i++) {
-      const { contractAddress } =
-        await deployFromAbiAndBytecodeForCreatorTxHash(
-          chainFixture.localSigner,
-          chainFixture.defaultContractArtifact.abi,
-          chainFixture.defaultContractArtifact.bytecode,
-          []
-        );
-      await chai
-        .request(serverFixture.server.app)
-        .post("/")
-        .field("address", contractAddress)
-        .field("chain", chainFixture.chainId)
-        .attach("files", chainFixture.defaultContractMetadata, "metadata.json")
-        .attach("files", chainFixture.defaultContractSource);
+  describe(`Pagination in /files/contracts/{full|any|partial}/${chainFixture.chainId}`, async function () {
+    const endpointMatchTypes = ["full", "any", "partial"];
+    for (const endpointMatchType of endpointMatchTypes) {
+      it(`should handle pagination in /files/contracts/${endpointMatchType}/${chainFixture.chainId}`, async function () {
+        const contractAddresses: string[] = [];
+
+        // Deploy 5 contracts
+        for (let i = 0; i < 5; i++) {
+          // Deploy partial matching contract if endpoint is partial or choose randomly if endpointMachtype is any. 'any' endpoint results should be consistent regardless.
+          const shouldDeployPartial =
+            endpointMatchType === "partial" ||
+            (endpointMatchType === "any" && Math.random() > 0.5);
```

Comment:
What's the purpose of using random instead of fixing for example odd as partial and even as full?

Reply:
No particular reason, could also use odd etc. The only purpose is to mix up partial and perfect matches.